### ISSUE TITLE
printf crashes when run without arguments

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#404]
Steps to reproduce: `printf`
Traceback:
```
Traceback (most recent call last):
  File "/usr/local/src/oil/bin/oil.py", line 887, in <module>
    sys.exit(main(sys.argv))
  File "/usr/local/src/oil/bin/oil.py", line 848, in main
    return AppBundleMain(argv)
  File "/usr/local/src/oil/bin/oil.py", line 818, in AppBundleMain
    status = ShellMain('osh', argv0, main_argv, login_shell)
  File "/usr/local/src/oil/bin/oil.py", line 615, in ShellMain
    status = main_loop.Interactive(opts, ex, c_parser, display, errfmt)
  File "/usr/local/src/oil/core/main_loop.py", line 90, in Interactive
    is_return, _ = ex.ExecuteAndCatch(node)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 1199, in ExecuteAndCatch
    status = self._Execute(node, fork_external=fork_external)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 1147, in _Execute
    status, check_errexit = self._Dispatch(node, fork_external)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 680, in _Dispatch
    status = self.RunSimpleCommand(arg_vec, fork_external)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 546, in RunSimpleCommand
    return self._RunBuiltin(builtin_id, arg_vec, fork_external)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 325, in _RunBuiltin
    status = self._RunBuiltinAndRaise(builtin_id, arg_vec, fork_external)
  File "/usr/local/src/oil/osh/cmd_exec.py", line 277, in _RunBuiltinAndRaise
    status = builtin_func(arg_vec)
  File "/usr/local/src/oil/osh/builtin_printf.py", line 145, in __call__
    fmt, fmt_spid = arg_r.ReadRequired2('requires a format string')
  File "/usr/local/src/oil/frontend/args.py", line 128, in ReadRequired2
    spid = self.spids[self.i]
IndexError: list index out of range
```

**Comment by andychu** [comment]
Oops, good find!

**Comment by andychu** [comment]
Released with 0.7.pre1: https://www.oilshell.org/release/0.7.pre1/
